odl/solvers/advanced/proximal_operators.py

```diff
@@ -0,0 +1,386 @@
+            to contain at least one operator.
+        ran : `ProductSpace`, optional
+            Range of the operator. If not provided, it is tried to be
+            inferred from the operators. This requires each **row**
+            to contain at least one operator.
+        """
+
+        indices = [range(len(operators)), range(len(operators))]
+        shape = (len(operators), len(operators))
+        op_matrix = sp.sparse.coo_matrix((operators, indices), shape)
+
+
+        return ProductSpaceOperator(op_matrix, dom=dom, ran=ran)
+
+    def make_diag(step_size):
+
+        return  diagonal_operator(
+            [factory(step_size) for factory in factory_list])
+
+    return make_diag
+
+
+def proximal_zero(space):
+    """Function to create the proximal operator of f(x) = 0.
+
+    Factory function which provides a function to initialize the proximal
+    operator of ``f(x) = 0`` where ``x`` is an element in ``space``. The
+    proximal operator of this functional is the identity operator.
+
+    Parameters
+    ----------
+    space : `DiscreteLp` or `ProductSpace` of `DiscreteLp`
+        Domain of the functional ``f(x)``
+
+    Returns
+    -------
+    make_prox : `function`
+        Function to initialize the proximal operator
+    """
+
+    def make_prox(tau):
+        """Return an instance of the proximal operator.
+
+        Parameters
+        ----------
+        tau : positive `float`
+            Step length parameter. Unused here but introduced to provide a
+            common interface
+
+        Returns
+        -------
+        id : `IdentityOperator`
+            The proximal operator instance of ``f(x) = 0`` which is the
+            identity operator
+        """
+
+        return IdentityOperator(space)
+
+    return make_prox
+
+
+# TODO: update doc
+def proximal_nonnegativity(space):
+    """Function to create the proximal operator of f(x) = 0.
+
+    Factory function which provides a function to initialize the proximal
+    operator of ``f(x)`` where ``x`` is an element in ``space``.
+
+    Parameters
+    ----------
+    space : `DiscreteLp` or `ProductSpace` of `DiscreteLp`
+        Domain of the functional ``f(x)``
+
+    Returns
+    -------
+    make_prox : `function`
+        Function to initialize the proximal operator
+    """
+
+    def make_prox(tau):
+        """Return an instance of the proximal operator.
+
+        Parameters
+        ----------
+        tau : positive `float`
+            Step length parameter. Unused here but introduced to provide a
+            common interface
+
+        Returns
+        -------
+        id : `Operator`
+            The proximal operator instance
+        """
+
+        class _ProxOpNonNegative(Operator):
+
+            """The proximal operator."""
+
+            def __init__(self):
+                """Initialize the proximal operator.
+
+                Parameters
+                ----------
+                tau : positive `float`
```

Comment:
Sure. But for the sake of unified interfaces (which we want to have in the class later anyway) it's better to add `tau` to the parameters and just not use it.